### PR TITLE
Update email sender https://wso2.org/jira/browse/ML-115

### PR DESCRIPTION
**Pull request by cdathuraliya**

### CODE REVIEW

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java [hunk 1]
```diff
@@ -599,6 +600,8 @@ public void run() {
                 MLModel model = modelBuilder.build();
                 
                 persistModel(id, ctxt.getModel().getName(), model);
+
+                emailTemplateParameters[1] = MLUtils.getLink(ctxt, MLConstants.MODEL_STATUS_COMPLETE);
                 EmailNotificationSender.sendModelBuildingCompleteNotification(emailNotificationEndpoint,
                         emailTemplateParameters);
             } catch (Exception e) {
```
Comment (cnapagoda) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java:607]: Can you please validate this exception handling?

Comment (nuwand) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java:607]: Its not a good idea to catch generic Exception. Its better to catch and handle the specific exception unless of course a third party API you depend on throws this Exception. In which case its better to comment and explain.

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLProjectHandler.java [hunk 2]
```diff
@@ -71,6 +71,14 @@ public MLProject getProject(int tenantId, String userName, String projectName) t
             throw new MLProjectHandlerException(e.getMessage(), e);
         }
     }
+
+    public MLProject getProject(int tenantId, String userName, long projectId) throws MLProjectHandlerException {
+        try {
+            return databaseService.getProject(tenantId, userName, projectId);
+        } catch (DatabaseHandlerException e) {
```
Comment (nuwand) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLProjectHandler.java:78]: Please log and throw the exception.

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/utils/MLUtils.java [hunk 3]
```diff
@@ -515,4 +520,48 @@ public static String getErrorMsg(String customMessage, Exception ex) {
...
+        DatabaseService databaseService = MLCoreServiceValueHolder.getInstance().getDatabaseService();
+        try {
+            analysis = databaseService.getAnalysis(tenantId, userName, analysisId);
+            analysisName = analysis.getName();
+            long projectId = analysis.getProjectId();
+
+            mlProject = databaseService.getProject(tenantId, userName, projectId);
+            projectName = mlProject.getName();
+            datasetId = mlProject.getDatasetId();
+        } catch (DatabaseHandlerException e) {
```
Comment (nuwand) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/utils/MLUtils.java:553]: Any reason for catching and throwing the same Exception?

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLProjectHandler.java (hunk 2 above)
Comment (nirmal070125) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLProjectHandler.java:78]: @nuwand this class is just a delegator and the DatabaseHandlerException is wrapped to MLProjectHandlerException. MLProjectHandlerException will be logged at the upstream and DatabaseHandlerException is logged at the downstream. Do we still need to log it here too? Isn't it just make it a long stack trace?

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java (hunk 1 above)
Comment (cdathuraliya) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java:607]: @cnapagoda @nuwand Fixed in pull request https://github.com/wso2/carbon-ml/pull/51. Thanks for pointing this out and for the explanation!

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/utils/MLUtils.java (hunk 3 above)
Comment (cdathuraliya) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/utils/MLUtils.java:553]: @nuwand Fixed in pull request https://github.com/wso2/carbon-ml/pull/51. Thanks for pointing this out!

**Pull request merged**